examples/sampling/graphbolt/node_classification.py

```diff
@@ -117,9 +117,14 @@ def create_dataloader(
     # [Role]:
     # Initialize a neighbor sampler for sampling the neighborhoods of nodes.
     ############################################################################
-    datapipe = datapipe.sample_neighbor(
-        graph, fanout if job != "infer" else [-1]
-    )
+    if args.sample_mode == "sample-neighbors":
+        datapipe = datapipe.sample_neighbor(
+            graph, fanout if job != "infer" else [-1]
+        )
+    else:
+        datapipe = datapipe.sample_layer_neighbor(
+            graph, fanout if job != "infer" else [-1]
+        )
```

Comment:
Do you think it would be better to have the following while enforcing `args.sample_mode` to be either `sample_neighbor` or `sample_layer_neighbor`?
```suggestion
    datapipe = getattr(datapipe, args.sample_mode)(
        graph, fanout if job != "infer" else [-1]
    )
```

Reply:
Or is this suggestion making the code too obscure?

Reply:
Because the args.sample_mode is `sample-neighbor` and the function is `sample_neighbor`. So using `getattr` may not work here.

Reply:
My suggestion below was making `-` into `_` so that `getattr` works.
https://github.com/dmlc/dgl/pull/7129#discussion_r1496229697

Reply:
OK, I will take this approach.